### PR TITLE
Setup improvements

### PR DESCRIPTION
**Pull request by TheSlimvReal**
- [x] Keycloak is required for new deployments
- [x] Merge compose files and use [profiles](https://docs.docker.com/compose/profiles/) to enable/disable backend
- [x] Automatically fetch latest app version from GitHub
- [x] Args can also be passed through command line
- [x] A Keycloak user is directly generated and a invitation mail is send out
- [x] Add instructions to infinitely listen to pipe and how to stop it
- [ ] Verify that no other commands can be run on server through pipe
- [x] Add default templates for common scenarios
- [x] Write setup script logs to 
- [x] Add assets to deployment
- [x] Default language

### CODE REVIEW

File: deployer/pipe-listener.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+# shellcheck disable=SC2046
+while true; do (cd /var/docker/setup; ./interactive_setup.sh $(cat deployer/arg-pipe) &> deployer/deploy-log.txt); done
```
Comment (sleidig) [inline deployer/pipe-listener.sh:3]: I am wondering if this script would be logically better kept in the deployer-backend project. But in the end it depends on both ndb-setup and deployer-backend of course, so I see why you have it here (also for easier execution)

File: docker-compose.yml [hunk 2]
```diff
@@ -7,16 +7,35 @@ services:
...
+      DATABASE_NAME: app
+      DATABASE_USER: admin
+      DATABASE_PASSWORD: ${COUCHDB_PASSWORD:?Admin password not set}
+      JWT_SECRET: ${JWT_SECRET:?JWT secret not set}
+      JWT_PUBLIC_KEY: "-----BEGIN PUBLIC KEY-----\n${PUBLIC_KEY}\n-----END PUBLIC KEY-----"
+      SENTRY_DSN: ${SENTRY_DSN}
+      PORT: 5984
     restart: unless-stopped
+    profiles:
+      - backend
```
Comment (sleidig) [inline docker-compose.yml:38]: nice! :+1:

File: docker-compose.yml [hunk 3]
```diff
@@ -7,16 +7,35 @@ services:
       - internal
       - nginx-proxy_default
     depends_on:
-      - couchdb
+      - ${COMPOSE_PROFILES:-couchdb}
```
Comment (sleidig) [inline docker-compose.yml:10]: what does this do?

File: deployer/pipe-listener.sh (hunk 1 above)
Comment (sleidig) [inline deployer/pipe-listener.sh:3]: Can we get our security expert to try and break this via the deployer-backend? I haven't invested time into possible hacks here yet.

**Review by sleidig — APPROVED** [review]
Looks good! Although I can't claim I re-checked every http request parameter or shell script arg mapping in the last detail.

From functional testing in combination with the deployer-backend + elementor plugin:
- What happens if using an existing sitename? Currently no feedback in the website form to users
- Email after signup:
  - subject + headline should contain “Aam Digital”
  - if possible, text should include a sentence about the new system being ready (not only talking about confirming email address)
- Keycloak display name would better be “Aam Digital - PROJECT”
- App:
  - no SiteSettings doc was available (clicking on it in menu also fails), but I do see it here, so maybe I wasn't testing on the most recent code
  - we should already set the name in SiteSettings (for the toolbar) to what the user selected (on [/self-signup](https://aam-digital.com/self-signup/)), I think it will say "Demo" otherwise.

otherwise functionality looks really nice!

File: deployer/pipe-listener.sh (hunk 1 above)
Comment (TheSlimvReal) [reply deployer/pipe-listener.sh:3]: As I think it has a much higher dependency on the `interactive-setup.sh` script, I prefer to have it here. In the end this is kind of a usage and other systems could also use this service with a different approach.

**Comment by TheSlimvReal** [comment]
@sleidig thanks for the extensive review.

- [x] Thanks for testing with the site name. It now shows a error message in the UI if this is the case
- [x] Keycloak display name is `Aam Digital - $org`
- [x] App site settings are present an name is `Aam Digital` (getting a custom site name in there is quite difficult)
- [x] Headline contains Aam Digital (due to display name)
- [ ] Subject contains Aam Digital -> Is that required as we have Aam Digital in the sender + in the message content? Other mails I get rarely have their name in the subject
- [ ] Adjust text -> quite difficult as Keycloak only comes with a limited set of actions and adding a [required action](https://www.keycloak.org/docs/latest/server_development/#required-action-walkthrough) is quite some work

**Review by sleidig — CHANGES_REQUESTED** [review]
Thanks for the updates :-)

From further testing:
- [x] site name accepts special characters (although it refuses space now) and then seems to break
- [ ] with the "error" in the end, it was loading a long time on the form. Can we display a message "Creating your system, this may take a few minutes ..." below the button once it's clicked?
- [ ] for the emails, I understand that customizing is too difficult. How about changing the email subject to "Your Aam Digital account" - which would be reasonable both for the first mail after registration as well as for password resets?

**Comment by TheSlimvReal** [comment]
@sleidig I improved the error handling and updated the email subject. For the submission I did not find a way to add some kind of progress spinner. We can try with a text on the submit button that indicates that it might take a little longer. Or if you have an idea how to improve this in elementor feel free to give it a try. There might be some fancy plugins out there just for this.

**Review by sleidig — APPROVED** [review]
Works really well now!

I have tweak the form in elementor a bit with styles and the German messages: https://aam-digital.com/codo-setup/

The new keycloak email subject was not deployed yet, I think. But overall great and good to merge :-)

**Pull request merged**